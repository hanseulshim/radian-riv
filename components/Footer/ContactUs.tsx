import React, { useState } from 'react'
import Input from 'components/common/Input'
import Modal from 'components/common/Modal'
import Form from 'components/common/Form'
import { validateEmail } from 'utils'

interface Props {
  closeModal: () => void
}

export default function ContactUs({ closeModal }: Props) {
  const toEmail = 'vow@redbellre.com'
  const [name, setName] = useState('')
  const [email, setEmail] = useState('')
  const [subject, setSubject] = useState('')
  const [address, setAddress] = useState('')
  const [comment, setComment] = useState('')
  const [emailError, setEmailError] = useState('')
  const [alert, setAlert] = useState(null)
  const submitForm = async () => {
    setAlert(null)
    const error = validateEmail(email)
    if (error) {
      setEmailError(error)
    } else {
      setAlert({ type: 'success', message: 'Email Sent' })
      const a = document.createElement('a')
      a.href = `mailto:${toEmail}?subject=${subject}&body=Name: ${name}%0D%0AEmail: ${email}%0D%0AProperty Address: ${address}%0D%0AComment: ${comment}`
      a.click()
    }
  }
  return (
    <Modal closeModal={closeModal} title={'Contact Us'} width={800}>
      <Form id="contact-us" onSubmit={submitForm} alert={alert}>
        <Input
          label="To"
          value={toEmail}
          error={''}
          onChange={() => {}}
          disabled
        />
        <div className="form-row">
          <div className="form-spacer">
            <Input
              label="Name"
              value={name}
              error={''}
              onChange={e => setName(e.target.value)}
            />
          </div>
          <div className="form-spacer">
            <Input
              label="Email"
              value={email}
              error={emailError}
              onChange={e => {
                setEmail(e.target.value)
                setEmailError('')
              }}
            />
          </div>
        </div>
        <Input
          label="Subject"
          value={subject}
          error={''}
          onChange={e => setSubject(e.target.value)}
        />
        <Input
          label="Property Address"
          value={address}
          error={''}
          onChange={e => setAddress(e.target.value)}
        />
        <textarea
          placeholder="Comment Field..."
          onChange={e => setComment(e.target.value)}
          value={comment}
        />
        <button className="btn" type="submit">
          Submit
        </button>
      </Form>
    </Modal>
  )
}
