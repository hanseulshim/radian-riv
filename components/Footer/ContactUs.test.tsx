import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import ContactUs from './ContactUs'

describe('Contact us', () => {
  let component
  beforeAll(() => {
    component = shallow(<ContactUs closeModal={() => {}} />)
  })
  it('should have form classes', () => {
    expect(component.find('.form-row')).toHaveLength(1)
    expect(component.find('.form-spacer')).toHaveLength(2)
  })
  it('should have text area', () => {
    expect(component.find('textarea').props().value).toEqual('')
    component
      .find('textarea')
      .simulate('change', { target: { value: '12345' } })
    expect(component.find('textarea').props().value).toEqual('12345')
    expect(component.find('textarea').props().placeholder).toEqual(
      'Comment Field...'
    )
  })
  it('should have submit button', () => {
    expect(component.find('button').text()).toEqual('Submit')
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
