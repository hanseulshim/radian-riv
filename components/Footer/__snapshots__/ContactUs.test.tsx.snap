// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact us should match snapshot 1`] = `
<Modal
  closeModal={[Function]}
  title="Contact Us"
  width={800}
>
  <Form
    alert={null}
    id="contact-us"
    onSubmit={[Function]}
  >
    <Input
      disabled={true}
      error=""
      label="To"
      onChange={[Function]}
      value="vow@redbellre.com"
    />
    <div
      className="form-row"
    >
      <div
        className="form-spacer"
      >
        <Input
          error=""
          label="Name"
          onChange={[Function]}
          value=""
        />
      </div>
      <div
        className="form-spacer"
      >
        <Input
          error=""
          label="Email"
          onChange={[Function]}
          value=""
        />
      </div>
    </div>
    <Input
      error=""
      label="Subject"
      onChange={[Function]}
      value=""
    />
    <Input
      error=""
      label="Property Address"
      onChange={[Function]}
      value=""
    />
    <textarea
      onChange={[Function]}
      placeholder="Comment Field..."
      value="12345"
    />
    <button
      className="btn"
      type="submit"
    >
      Submit
    </button>
  </Form>
</Modal>
`;
